src/hooks/useCryptoBalances.ts

```diff
@@ -0,0 +1,184 @@
+import { useMemo } from 'react'
+
+import {
+  AcceptedToken,
+  getChainId,
+  getTokenAddress,
+  SupportedChain,
+} from '@/types/chains'
+import { CRYPTO_CONFIG } from '@/utils/walletConfig'
+
+import { useCryptoBalance } from './useCryptoBalance'
+
+interface UseCryptoBalancesProps {
+  selectedNetwork: string
+}
+
+export const useCryptoBalances = ({
+  selectedNetwork,
+}: UseCryptoBalancesProps) => {
+  // Fetch balances for each crypto asset
+  const cusdCeloBalance = useCryptoBalance(
+    getTokenAddress(SupportedChain.CELO, AcceptedToken.CUSD),
+    getChainId(SupportedChain.CELO)
+  )
+  const usdcCeloBalance = useCryptoBalance(
+    getTokenAddress(SupportedChain.CELO, AcceptedToken.USDC),
+    getChainId(SupportedChain.CELO)
+  )
+  const usdtCeloBalance = useCryptoBalance(
+    getTokenAddress(SupportedChain.CELO, AcceptedToken.USDT),
+    getChainId(SupportedChain.CELO)
+  )
+
+  const usdcArbitrumBalance = useCryptoBalance(
+    getTokenAddress(SupportedChain.ARBITRUM, AcceptedToken.USDC),
+    getChainId(SupportedChain.ARBITRUM)
+  )
+  const usdtArbitrumBalance = useCryptoBalance(
+    getTokenAddress(SupportedChain.ARBITRUM, AcceptedToken.USDT),
+    getChainId(SupportedChain.ARBITRUM)
+  )
+
+  const usdcArbitrumSepoliaBalance = useCryptoBalance(
```

Comment:
We want to expand to more chains in the future, so having a manual approach would snowball into a lot of code quickly. What we want is a way to define just the chain and pull the token information based on what we have defined in the supportedChains variables. We also don't want to need to wait for 6 variables to do the same thing. We should instead run all these fetches in parallel by having a functional-based approach